### PR TITLE
handled error if vlan is passed without ovs container being available

### PR DESCRIPTION
**Pull request by abdulrahmantkhalifa**
this will fix issue #765

### CODE REVIEW

File: api/node/node_api_UpdateContainer.go [hunk 1]
```diff
@@ -48,6 +53,20 @@ func (api NodeAPI) UpdateContainer(w http.ResponseWriter, r *http.Request) {
 			tools.WriteError(w, http.StatusBadRequest, err, "")
 			return
 		}
+		if nic.Type == EnumContainerNICTypevlan {
```
Comment (muhamadazmy) [inline api/node/node_api_UpdateContainer.go]: What if it's a vxlan type ?

Comment (grimpy) [inline api/node/node_api_UpdateContainer.go]: indeed should be a check for both vxlan and vlan

File: api/node/node_api_UpdateContainer.go [hunk 2]
```diff
@@ -48,6 +53,20 @@ func (api NodeAPI) UpdateContainer(w http.ResponseWriter, r *http.Request) {
...
+			container := client.Container(cl)
+			tags := []string{"ovs"}
+			containers, err := container.Find(tags)
```
Comment (muhamadazmy) [inline api/node/node_api_UpdateContainer.go]: Take the find outside of the loop

File: api/tools/ays.go [hunk 3]
```diff
@@ -114,6 +114,17 @@ func (aystool AYStool) WaitJobDone(jobid, repoName string) (ays.Job, error) {
 			return job, err
 		}
 	}
+
+	if job.State == "error" {
+		msg := ""
+		for _, log := range job.Logs {
+			if log.Category == "errormsg" {
+				msg = log.Log
```
Comment (grimpy) [inline api/tools/ays.go]: can return at this line and remove the variable msg

File: templates/container/actions.py [hunk 4]
```diff
@@ -173,7 +173,10 @@ def get_nic_index(nic):
         if get_nic_id(nic_dict) not in ids_current_nics:
             nic_dict.pop('token', None)
             logger.info("Adding nic to container {}: {}".format(container.id, nic_dict))
-            cl.nic_add(container.id, nic_dict)
+            try:
+                cl.nic_add(container.id, nic_dict)
+            except Exception as e:
+                raise j.exceptions.Input(e.__str__())
```
Comment (grimpy) [inline templates/container/actions.py]: we are going to implemetn errocodes in the erros of core0 please us this

**Comment by abdulrahmantkhalifa** [comment]
depends on https://github.com/Jumpscale/ays9/pull/182

File: api/tools/ays.go [hunk 5]
```diff
@@ -101,7 +107,7 @@ func (aystool AYStool) WaitRunDone(runid, repoName string) (*ays.AYSRun, error)
 
 func (aystool AYStool) WaitJobDone(jobid, repoName string) (ays.Job, error) {
 	job, resp, err := aystool.Ays.GetJob(jobid, repoName, nil, nil)
-
+	fmt.Print(resp)
```
Comment (khaledkbadr) [inline api/tools/ays.go]: remove this, or use logger if logging was the intention here

File: api/vendor/github.com/zero-os/0-core/client/go-client/container.go [hunk 6]
```diff
@@ -68,6 +69,16 @@ func (c *containerMgr) List() (map[int16]ContainerResult, error) {
 	return result, nil
 }
 
+func (c *containerMgr) Find(tags Tags) (map[int16]ContainerResult, error) {
```
Comment (khaledkbadr) [inline api/vendor/github.com/zero-os/0-core/client/go-client/container.go]: if not implemented in core-0 remove

File: api/tools/ays.go (hunk 5 above)
Comment (abdulrahmantkhalifa) [inline api/tools/ays.go]: debuggin reminants, srry

File: api/tools/ays.go [hunk 7]
```diff
@@ -114,6 +119,21 @@ func (aystool AYStool) WaitJobDone(jobid, repoName string) (ays.Job, error) {
 			return job, err
 		}
 	}
+
+	if job.State == "error" {
+		err := AYSError{}
+		json.Unmarshal([]byte(job.Result), &err)
```
Comment (grimpy) [inline api/tools/ays.go]: Unmarshal returns error which is not checked which could give a panic on the next line

File: templates/container/actions.py [hunk 8]
```diff
@@ -173,7 +177,15 @@ def get_nic_index(nic):
         if get_nic_id(nic_dict) not in ids_current_nics:
             nic_dict.pop('token', None)
             logger.info("Adding nic to container {}: {}".format(container.id, nic_dict))
-            cl.nic_add(container.id, nic_dict)
+            try:
+                cl.nic_add(container.id, nic_dict)
+            except RuntimeError as e:
+                if len(e.args) >= 2:
+                    core_job = e.args[1]
+                if 499 >= core_job.code >= 400:
```
Comment (grimpy) [inline templates/container/actions.py]: core_job is undefined if len e.args = 1

**Pull request merged**